lib/board.rb

```diff
@@ -0,0 +1,58 @@
+class Board
+
+  attr_accessor :cells
+
+def initialize (cells = nil)
+  @cells = cells || Array.new(9, " ")
+end
+
+def display
+  puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+  puts "-----------"
+  puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+  puts "-----------"
+  puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+end
+
+def reset!
+  self.cells = Array.new(9, " ")
+end
+
+def position(index)
+  cell = input_to_index(index)
+  self.cells[cell]
+end
+
+def input_to_index(input)
+  index = input.to_i - 1
+end
+
+def full?
+   self.cells.none? {|cell| cell.lstrip.empty?}
+end
+
+def turn_count
+  self.cells.select{|c| !c.lstrip.empty?}.count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```